prow/rewriteConfig.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+)
+
+// Count the number of spaces before a valid character in a string.
+func countLeadingSpace(line string) int {
+	i := 0
+	for _, runeValue := range line {
+		if runeValue == ' ' {
+			i++
+		} else {
+			break
+		}
+	}
+	return i
+}
+
+// Add element to the slice at the indexed position and move all the strings from
+// the index one position to the right.
+func appendAtIndex(original []string, element string, index int) []string {
+	original = append(original, "")
+	copy(original[index+1:], original[index:])
+	original[index] = element
+
+	return original
+}
+
+// Copy the content of one slice to another.
+func copySlice(input []string) []string {
+	newSlice := []string{}
+	newSlice = append(newSlice, input...)
+
+	return newSlice
+}
+
+// Find index of master section in each branch split for the new branch and
+// get content from master section with mandatory "merges-blocked-needs-admin" part.
+func getContentOfNewBranch(branchLines []string, newBranch string) (int, string) {
```

Comment:
This looks extremely complex.
will it work if you find  "  master:" string, count number of spaces for master and find next line with this number of leading spaces?

Is there anything else which need to be done? Am I missing something?